server/lib/order/order.ctrl.js

```diff
@@ -0,0 +1,165 @@
+const keygen = require('generate-key')
+
+const getNewAddress = require('../rpc/get-new-address')
+const ChangellyCtrl = require('../changelly/changelly.ctrl')
+const TransactionCtrl = require('../db/transaction.ctrl')
+
+const OrderCtrl = {}
+
+OrderCtrl.createOrder = (req, res) => {
+  OrderCtrl.validateParams(req)
+  .then(() => {
+    OrderCtrl.beginOrderCreation(req, res)
+  })
+  .catch((error) => {
+    OrderCtrl.handleError(error, res, '002')
+  })
+}
+
+OrderCtrl.beginOrderCreation = (req, res) => {
+  OrderCtrl.getNavAddress()
+  .then((address) => {
+    req.params.navAddress = address
+    OrderCtrl.getFirstChangellyAddress(req, res)
```

Comment:
we should probably look to decouple the parameters from the req to make the code more modular if we need to make changes to the req structure in the future. Again, we can refactor this later.